arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -1,42 +1,128 @@
 package arrow.test.laws
 
 import arrow.Kind
+import arrow.core.Either
 import arrow.core.Eval
+import arrow.core.Eval.Companion.always
+import arrow.core.ForId
+import arrow.core.ForListK
 import arrow.core.Id
+import arrow.core.Left
+import arrow.core.ListK
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Right
+import arrow.core.Some
+import arrow.core.extensions.ListKEq
+import arrow.core.extensions.either.monad.flatMap
 import arrow.core.extensions.eq
+import arrow.core.extensions.id.applicative.applicative
 import arrow.core.extensions.id.comonad.extract
+import arrow.core.extensions.id.eqK.eqK
 import arrow.core.extensions.id.monad.monad
+import arrow.core.extensions.id.traverse.traverse
+import arrow.core.extensions.list.eqK.eqK
+import arrow.core.extensions.listk.eq.eq
+import arrow.core.extensions.listk.monoid.monoid
 import arrow.core.extensions.monoid
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.extensions.option.eq.eq
+import arrow.core.extensions.option.eqK.eqK
+import arrow.core.extensions.option.traverseFilter.traverseFilter
+import arrow.core.identity
+import arrow.core.right
+import arrow.core.some
 import arrow.test.concurrency.SideEffect
 import arrow.test.generators.GenK
+import arrow.test.generators.functionAAToA
+import arrow.test.generators.functionABToB
 import arrow.test.generators.functionAToB
+import arrow.test.generators.functionBAToB
+import arrow.test.generators.genEval
+import arrow.test.generators.genK
 import arrow.test.generators.intPredicate
 import arrow.test.generators.intSmall
+import arrow.test.generators.listK
+import arrow.typeclasses.Applicative
 import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
 import arrow.typeclasses.Foldable
 import io.kotlintest.properties.Gen
 import io.kotlintest.properties.forAll
 
 object FoldableLaws {
 
   fun <F> laws(FF: Foldable<F>, GENK: GenK<F>): List<Law> {
-    val GEN = GENK.genK(Gen.intSmall())
-    val EQ = Int.eq()
+    val GEN: Gen<Kind<F, Int>> = GENK.genK(Gen.intSmall())
+    val GENListK: Gen<Kind<F, ListK<Int>>> = GENK.genK(Gen.listK(Gen.intSmall()))
+    val GENB = GENK.genK(Gen.bool())
 
-    return listOf(
+    val EQ: Eq<Int> = Int.eq()
+    val EQBool = Boolean.eq()
+    val EQOptionInt = Option.eq(Int.eq())
+    val EQListKInt: ListKEq<Int> = ListK.eq(Int.eq())
+    val EQForIdInt: Eq<Kind<ForId, Int>> = Id.eqK().liftEq(Int.eq())
+
+    return laws2(Id.traverse(), Id.applicative(), Id.genK(), Id.eqK()) +
```

Comment:
This is pointless. `laws2` does not need to have `Foldable` and `Applicative` for `F`. `traverse_` and `sequence_` work with any applicative. So use `Foldable<F>` (`FF`) and any applicative for example `Id`. Also please inline `laws2`, it does not need to be seperate.

Reply:
Hi @1Jajen1 I will need some help to do that because I don't see the way to do it. If I use an instance of applicative Id.. I will have to change the generated function and Eq to check `ForId` and not any `F`. Thats why they were not inline within all laws together. Any hint is welcome!

Reply:
`fun <F, G> Foldable<F>.traverse_Test(AP: Applicative<G>, GF: GenK<F>, GG: GenK<G>, EQF: EqK<F>, EQG: EqK<G>)` and then basically the test you added below with data generated by these generators and validated by the eq instances.
In the laws just invoke them with any applicative of your choice. `traverseTest(Option.applicative(), GENK, Option.genK(), EQK, Option.eqK())`.

Basically just get rid of the constraint that for invoking `traverse_` you need the applicative and foldable for the **same** `F`. That is simply not the case.

Reply:
Thanks @1Jajen1 , will try this 👍

Reply:
Made the change, how does it look like now?